### PR TITLE
Rename pointer data size type contiguous

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

File: include/experimental/__p0009_bits/config.hpp [hunk 1]
```diff
@@ -279,7 +279,7 @@ static_assert(_MDSPAN_CPLUSPLUS >= MDSPAN_CXX_STD_14, "mdspan requires C++14 or
 // Corentins demo compiler for subscript chokes on empty [] call,
```
Comment (mhoemmen) [inline include/experimental/__p0009_bits/config.hpp:279]:
```suggestion
// Early implementations of P2128 (multiple-parameter operator[]) fail to compile
// if called with zero arguments, even though the proposal supports this.
```

File: include/experimental/__p0009_bits/config.hpp [hunk 2]
```diff
@@ -279,7 +279,7 @@ static_assert(_MDSPAN_CPLUSPLUS >= MDSPAN_CXX_STD_14, "mdspan requires C++14 or
...
 // though I believe the proposal supports it?
```
Comment (mhoemmen) [inline include/experimental/__p0009_bits/config.hpp:280]:
```suggestion
```

**Review by mhoemmen — DISMISSED** [review]

**Comment by mhoemmen** [comment]
nvc++'s build works if I rebase this PR atop my PR #148.  I get one build warning (which explains the one loop change in PR #148):
```
[ 92%] Building CXX object tests/CMakeFiles/test_mdarray_ctors.dir/test_mdarray_ctors.cpp.o
".../include/experimental/__p0009_bits/layout_right.hpp", line 174: warning: pointless comparison of unsigned integer with zero
        for(rank_type r=0; r<extents_type::rank(); r++) value*=__extents.extent(r);
                            ^
          detected during:
            instantiation of "std::experimental::layout_right::mapping<Extents>::index_type std::experimental::layout_right::mapping<Extents>::required_span_size() const noexcept [with Extents=std::experimental::extents<unsigned int>]" at line 237 of ".../include/experimental/__p1684_bits/mdarray.hpp"
            instantiation of "std::experimental::mdarray<ElementType, Extents, LayoutPolicy, Container>::mdarray(const std::experimental::mdarray<ElementType, Extents, LayoutPolicy, Container>::extents_type &, const Alloc &) [with ElementType=int, Extents=std::experimental::extents<unsigned int>, LayoutPolicy=std::experimental::layout_right, Container=std::vector<int, std::allocator<int>>, Alloc=std::allocator<int>, <unnamed>=0]" at line 365 of ".../tests/test_mdarray_ctors.cpp"
```

**Comment by mhoemmen** [comment]
I'm not sure why github is more conservative about conflicts than git.  Rebasing #147 atop #149 worked for me.

**Comment by crtrott** [comment]
Because it requires a force push due to rewriting history I guess.

**Review by mhoemmen — APPROVED** [review]
Thanks @crtrott ! : - D  It builds for me with nvc++ and passes all the tests.

**Pull request merged**